system-addon/content-src/components/TopSites/TopSitesEdit.jsx

```diff
@@ -73,7 +73,19 @@ export class _TopSitesEdit extends React.PureComponent {
     let editIndex = this.state.editIndex;
     if (showEditForm && this.props.TopSites.editForm.visible) {
       const targetURL = this.props.TopSites.editForm.site.url;
-      editIndex = this.props.TopSites.rows.findIndex(s => s.url === targetURL);
+      editIndex = this.props.TopSites.rows.findIndex(s => s && s.url === targetURL);
```

Comment:
Could we just always rely on getting an index instead of searching for something by url? The link menu option for `EditTopSite` should be able to get the `index` just like `PinTopSite` ? https://github.com/mozilla/activity-stream/blob/f0f1daa8e0d1155138025b0a13d3e7a6a3cd73da/system-addon/content-src/lib/link-menu-options.js#L78